src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -475,3 +484,44 @@ bool java_bytecode_languaget::to_expr(
 java_bytecode_languaget::~java_bytecode_languaget()
 {
 }
+
+/// Activate the generic instantiation code.
+/// \param message_handler
+/// \param symbol_table The symbol table so far.
+void
+java_bytecode_languaget::handle_generics(
```

Comment:
I think my preference would be to put this method in `generate_java_generic_type.h` and probably call it something more descriptive (since `handle_generics` is both vague (handle how?) and wrong (doesn't solve all generic problems :stuck_out_tongue: ))

Reply:
My thought process for this was for it to be locus of control for all handling of generics in the `java_bytecode_languaget` class, containing the whole sequence of steps (not just mine) that are required for handling generics, and then just hook this self contained unit into whatever step of the pipeline of the conversion we want, be it after type-checking, or after conversion, or whatever.
Therefore, I do not think it's appropriate for it to go into `generate_java_generic_type.h`, unless its semantics change to the point where it only does the adding to the symbol table and nothing else.

To put it in fewer words, it's just a re-organisation of the original logic that I had, for it to be more modular.